app/src/main/java/xyz/jhughes/laundry/apiclient/MachineAPI.java

```diff
@@ -1,39 +1,15 @@
 package xyz.jhughes.laundry.apiclient;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Map;
-
-import retrofit2.Call;
-import retrofit2.http.GET;
-import retrofit2.http.Path;
-import xyz.jhughes.laundry.LaundryParser.Locations;
-import xyz.jhughes.laundry.LaundryParser.Machine;
-import xyz.jhughes.laundry.LaundryParser.MachineList;
 
 /**
- * Retrofit interface for the Machine API.
+ * Singleton impl for the Machine API.
  */
-public interface MachineAPI {
-    @GET("/v2/location/{location}")
-    Call<ArrayList<Machine>> getMachineStatus(
-            @Path("location") String location
-    );
-
-    @GET("/v2/location/all")
-    Call<Map<String,MachineList>> getAllMachines();
-
-    @GET("/v2/locations")
-    Call<List<Locations>> getLocations();
-
-    @GET("/v2-debug/location/{location}")
-    Call<ArrayList<Machine>> getMachineStatus_DEBUG(
-            @Path("location") String location
-    );
+public class MachineAPI {
```

Comment:
This file would have the retrofit request to the server. I can either do that in this PR or in a separate one but we will probably have to bring in rxjava if i am to do it now

Reply:
I added the methods for getting the Retrofit callback.